coil-compose-base/src/main/java/coil/compose/AsyncImagePainter.kt

```diff
@@ -295,14 +301,14 @@ class AsyncImagePainter internal constructor(
 
         // If we're in inspection mode skip the image request and set the state to loading.
         if (isPreview) {
-            val request = request.newBuilder().defaults(imageLoader.defaults).build()
+            val request = requestInternal.value.newBuilder().defaults(imageLoader.defaults).build()
             updateState(State.Loading(request.placeholder?.toPainter()))
-            return
+            return@trace
         }
 
         // Observe the current request and execute any emissions.
         scope.launch {
-            snapshotFlow { request }
+            requestInternal
                 .mapLatest { imageLoader.execute(updateRequest(it)).toState() }
```

Comment:
What if this is actually just a `var _request: ImageRequest`?
The request is changed from outside right before calling `onRemembered` manually, which means it's actually up-to-date here.

I don't know what's the case for the nested painters.

Reply:
I think we might need to read this as a state, but I can't remember the specifics. If I remember correctly there are cases were `mutableStateOf(request)` won't be equal to `_request` since it exists outside of the Composition. Also it's possible for `request` to change so we need to observe it.